src/dotnet/MsbuildProject.cs

```diff
@@ -62,6 +62,19 @@ public static MsbuildProject FromFile(ProjectCollection projects, string project
 
         public static MsbuildProject FromDirectory(ProjectCollection projects, string projectDirectory)
         {
+            FileInfo projectFile = GetProjectFileFromDirectory(projectDirectory);
+
+            var project = TryOpenProject(projects, projectFile.FullName);
```

Comment:
I know you didn't right this API, so don't worry about it, but this is a weird API. It says TryOpenProject. I associate TryXXX APIs with APIs that return bool and an out param. Yet, this one returns null when it fails.

@krwq In case you wrote this and would like to address this at some point.